### PR TITLE
build: checkdeps only if relevant changes were made

### PR DESCRIPTION
**Pull request by tamird**
I'm going to push a second commit that should trigger the failure case.

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/14477)

### CODE REVIEW

File: pkg/cmd/github-pull-request-make/main.go [hunk 1]
```diff
@@ -186,6 +186,40 @@ func main() {
...
+				log.Fatal(err)
+			}
+		}
+
+		// Check for diffs.
+		for _, dir := range []string{crdb.Dir, filepath.Join(crdb.Dir, "vendor")} {
+			cmd := exec.Command("git", "diff")
+			cmd.Dir = dir
+			log.Println(cmd.Args)
+			if output, err := cmd.Output(); err != nil {
```
Comment (dt) [inline pkg/cmd/github-pull-request-make/main.go]: I might find this error a little easier to understand and fix if it logged a bit more explanation / context.

**Review by dt — APPROVED** [review]

**Comment by tamird** [comment]
Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[pkg/cmd/github-pull-request-make/main.go, line 215 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/14477#-KgVF91IWRZ9kK12gKNw-r2-215:-KgVUmeHOS-HNMmIeT8V:b-h7imtx) ([raw file](https://github.com/cockroachdb/cockroach/blob/177a90d0ba3542389bf519219806e02c6279978d/pkg/cmd/github-pull-request-make/main.go#L215)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

I might find this error a little easier to understand and fix if it logged a bit more explanation / context.
</blockquote></details>

Done. The error should be self-explanatory; the command is logged before it runs.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14477)*

**Comment by dt** [comment]
I have a slight preference to add a flag (or TARGET value) that causes `github-pull-request-make` to just check vendor and then exit, rather than adding the check to the other builds, but I don't feel especially strongly.

**Comment by tamird** [comment]
Done.

**Review by dt — APPROVED** [review]

**Review by dt — APPROVED** [review]

**Pull request merged**